libs/x-content/src/main/java/org/elasticsearch/xcontent/Text.java

```diff
@@ -36,31 +31,45 @@ public static Text[] convertFromStringArray(String[] strings) {
         return texts;
     }
 
-    private BytesReference bytes;
+    private EncodedBytes bytes;
     private String text;
     private int hash;
+    private int stringLength = -1;
+
+    /**
+     * Construct a Text from a UTF-8 encoded ByteBuffer. Since no string length is specified, {@link #stringLength()}
```

Comment:
Is this comment correct? It's not using a `ByteBuffer` now.

Also I wonder if the fact that it's UTF-8 should be reflected somewhere in the naming? Perhaps `EncodedBytes` could be `Utf8Bytes`? I'm not sure whether that's appropriate but it's a thought.

Reply:
`UTF8Bytes` makes sense to me! I'll rename it

Reply:
Looking at the implementation of `EncodedBytes.equals` and `compareTo`, that class is aware that it's UTF-8, so it's probably appropriate to call it `Utf8Bytes` or similar.

Reply:
I'm still leaning towards calling it `UTF8Bytes` because its main purpose is as a return type from `XContentString#bytes()`, which specifies in the javadoc that it's always UTF-8 encoded.